python/cuml/preprocessing/model_selection.py

```diff
@@ -120,30 +132,106 @@ def train_test_split(
                 "proportion train_size should be between"
                 "0 and 1 (found {})".format(train_size)
             )
-        split_idx = int(X.shape[0] * train_size)
 
     if isinstance(train_size, int):
         if not 0 <= train_size <= X.shape[0]:
             raise ValueError(
                 "Number of instances train_size should be between 0 and the"
                 "first dimension of X (found {})".format(train_size)
             )
-        split_idx = train_size
 
-    if seed is not None:
-        np.random.seed(seed)
+    if isinstance(test_size, float):
+        if not 0 <= test_size <= 1:
+            raise ValueError(
+                "proportion test_size should be between"
+                "0 and 1 (found {})".format(train_size)
+            )
 
-    # Replace Numpy/cuDF here when issue mentioned above is solved!
+    if isinstance(test_size, int):
+        if not 0 <= test_size <= X.shape[0]:
+            raise ValueError(
+                "Number of instances test_size should be between 0 and the"
+                "first dimension of X (found {})".format(test_size)
+            )
+
+    x_numba = False
+    y_numba = False
     if shuffle:
-        idxs = np.arange(len(X))
-        _shuffle_idx(idxs)
-        X = X.iloc[idxs].reset_index(drop=True)
-        y = y.iloc[idxs].reset_index(drop=True)
-
-    split_idx = int(X.shape[0] * train_size)
-    X_train = X.iloc[0:split_idx]
-    y_train = y.iloc[0:split_idx]
-    X_test = X.iloc[split_idx:]
-    y_test = y.iloc[split_idx:]
+        if seed is None or isinstance(seed, int):
```

Comment:
Don't need to make a utility yet, but I could see this being reused as we build out more Python-based algorithms.

Reply:
I was thinking the same thing, was writing a prototype already :)